sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Statsbeat/Statsbeat.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics.Metrics;
+using System.Threading;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Internals
+{
+    internal static class Statsbeat
+    {
+        private static readonly Meter s_myMeter = new("AttachStatsBeatMeter", "1.0");
+
+        private const string StatsBeat_ConnectionString = "<StatsBeat_ConnectionString>";
+
+        // TODO: Move IsWindowsOS() to new class
+        // Do we need to support OSX?
+        internal static string s_OS = StorageHelper.IsWindowsOS() ? "windows" : "linux";
+
+        internal const int AttachStatsBeatInterval = 86400000;
+
+        static Statsbeat()
+        {
+            s_myMeter.CreateObservableGauge("AttachStatsBeat", () => GetAttachStatsBeat());
+
+            // Configure for attach statsbeat which has collection
+            // schedule of 24 hrs == 86400000 milliseconds.
+            Sdk.CreateMeterProviderBuilder()
+            .AddMeter("AttachStatsBeatMeter")
+            .AddStatsbeatExporter(o =>
```

Comment:
can we not use AzMonExporter itself?

Reply:
We might be able to but that will need some changes on AzMonExporter side like special casing for resources. For statsbeat I introduced new properties on `ResourceParser` and AzMonexporter does not allow export interval to be configurable right now. Also, if we need to make any changes to just statsbeat, having separate exporter might help better in that case.

Reply:
> like special casing for resources

Not clear why. Can you elaborate?

Reply:
> AzMonexporter does not allow export interval

Can you modify the Gauge callback to deal with it? Return measurement only once every 15 mins, else return empty?

Reply:
> > AzMonexporter does not allow export interval
> 
> Can you modify the Gauge callback to deal with it? Return measurement only once every 15 mins, else return empty?

I am not sure if I understood this - do you have an example?

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Instrumentation.Runtime/RuntimeMetrics.cs#L65-L74

Reply:
Added internal option to deal with this.